packages/@ember/runloop/index.js

```diff
@@ -26,6 +27,7 @@ let flush;
 if (EMBER_METAL_TRACKED_PROPERTIES) {
   flush = function(queueName, next) {
     if (queueName === 'render' || queueName === _rsvpErrorQueue) {
+      flushSyncObservers();
```

Comment:
Can you add a test specifically for this (e.g. unrelated to query params)?

Reply:
Discussed this with @wycats and @chancancode today actually, and it turns out we don't need to do this for this PR. I'm going to open up a followup with it and the tests.

Reply:
Seems good. Though I _do_ think we need to do this specific thing (guarantee that the sync observers are flushed).